compiler/rustc_hir_pretty/src/lib.rs

```diff
@@ -1367,7 +1367,7 @@ impl<'a> State<'a> {
         self.ann.pre(self, AnnNode::Expr(expr));
         match expr.kind {
             hir::ExprKind::Box(expr) => {
-                self.word_space("box");
+                self.word_space("#[rustc_box]");
```

Comment:
You're missing a `Box::new()` here :)

Reply:
Since the attr is printed anyway, switched this out for a `Box::new` + `print_call_post` with the content